app/tests/unit/controllers/object.spec.js

```diff
@@ -0,0 +1,112 @@
+const Problem = require('api-problem');
+const { AuthType } = require('../../../src/components/constants');
+
+const controller = require('../../../src/controllers/object');
+const { storageService, objectService, versionService } = require('../../../src/services');
+
+const mockResponse = () => {
+  const res = {};
+  res.status = jest.fn().mockReturnValue(res);
+  res.json = jest.fn().mockReturnValue(res);
+  return res;
+};
+
+// Mock config library - @see {@link https://stackoverflow.com/a/64819698}
+jest.mock('config');
+
+const storageDeleteObjectSpy = jest.spyOn(storageService, 'deleteObject');
+const objectDeleteSpy = jest.spyOn(objectService, 'delete');
+const versionCreateSpy = jest.spyOn(versionService, 'create');
+const versionDeleteSpy = jest.spyOn(versionService, 'delete');
+const versionListSpy = jest.spyOn(versionService, 'list');
+
+
+const res = mockResponse();
```

Comment:
In general, it's best practice to not declare spies and mock constructs as top-level objects in a Jest test; values in general should only be declared at the most specific block scope possible.

These values should be declared under the `describe('deleteObject', () => {` block scope. (Reinsert to Line 28)

Also when using spyOn, you generally want an associated `beforeEach()` lifecycle that will `mockReset()` them in order to guarantee test atomicity. Without this, you may be testing behavior with unintentional side effects.